modules/channel-web/src/views/lite/typings.d.ts

```diff
@@ -247,6 +247,15 @@ interface CustomButton {
   onClick: (buttonId: string, headerComponent: JSX.Element, event: React.MouseEvent) => void
 }
 
+interface CustomAction {
+  /** An ID to identify your action. It is required to remove it */
+  id: string
+  /** Text rendered in the context menu */
+  text: string
+  /** Handler called when clicking the action in the context menu */
+  select: (eventId: string) => void
+}
```

Comment:
To keep consistency, can you use 'label' for the text, and onClick instead of select ?

Reply:
Yeah label is fine, but I don't believe this function should be called onClick because an onClick function should have an event as first parameter and this one only receive a string.

Reply:
addHeaderButton and addContextActions are almost identical and should share a common signature. Otherwise, it complexifies documentation and uses different ways to do the same thing. 

The only thing which would change from CustomButton to CustomAction is the second parameter (headerComponent), which would be the MessageComponent

Reply:
Which MessageComponent should I pass? I don't have any component in my hands to pass except the MenuItem component. I can change the signature for:

onClick: (eventId: string, actionId: string, event: React.MouseEvent) => void

Reply:
The signature of CustomButton looks like this: 
```js
onClick={btn.onClick.bind(this, btn.id, this)}
```
So it sends the button ID, the current component, then the mouse event.
If we apply the same signature, it would be buttonId, messageProps, mouseEvent 

If you'd like to set the message props as first element, we should then also update the signature of custombuttons so it's identical